ui/src/util/basePaths.ts

```diff
@@ -1,25 +1,18 @@
 import { removeTrailingSlash } from "util/removeTrailingSlash";
+import { getFullPath } from "./getFullPath";
 type BasePath = `/${string}`;
 
-declare global {
-  interface Window {
-    base?: string;
-  }
-}
-
 export const calculateBasePath = (): BasePath => {
-  let basePath = "";
-  if ("base" in window && typeof window.base === "string") {
-    basePath = window.base;
-  }
-  if (basePath) {
-    return `${removeTrailingSlash(basePath)}/` as BasePath;
+  const basePath = document.querySelector("base")?.href;
+  const path = basePath ? getFullPath(basePath) : null;
+  if (path) {
+    return `${removeTrailingSlash(path)}/` as BasePath;
   }
-  return "/";
+  return "/ui/";
 };
 
-export const basePath: BasePath = `${calculateBasePath()}ui`;
-export const apiBasePath: BasePath = `${calculateBasePath()}api/v0/`;
+export const basePath: BasePath = calculateBasePath();
+export const apiBasePath: BasePath = "/api/v0/";
```

Comment:
Looking at the backed the API is always at the root and isn't prefixed by the context path like the UI is. Is that correct?

Reply:
No, when running behind a proxy the backend may be served under a path. This is how our charmed traefik works.

The only thing you know is that the backend is served on `{basePath}/..` (because `basepath == "https://path/to/backend/ui`). On top of that there is the convention that all our handlers are served under `api`

Reply:
Exactly, it is always been like that.
To sum up
You can't run the admin-ui executable APIs under a context path (on local machine for example).
But when deploying, traefik proxy will handle the request under a context path, letting admin ui "believe" it's under no path at all.